make-uniform/production/Build Standard Database.Rmd

```diff
@@ -10,53 +10,77 @@ output:
 ## Administration
 
 #### Purpose
-Procedure to translate any number of on-board survey data sets into a single dataset with common variables and common responses.  This is a work-in-progress production version of the procedures.  See ``../mini-example`` and ``../small-example`` for working proto-types.  In this script, we put in place procedures to process surveys into a standard database.  See ``Extract Variables from Legacy Surveys.Rmd`` for procedures to extract variables from legacy surveys (i.e., those with SAS summary scripts). 
+Procedure to translate any number of on-board survey data sets into a single 
+dataset with common variables and common responses.  This is a work-in-progress 
+production version of the procedures.  See ``../mini-example`` and 
+``../small-example`` for working proto-types.  In this script, we put in place 
+procedures to process surveys into a standard database.  See 
+``Extract Variables from Legacy Surveys.Rmd`` for procedures to extract 
+variables from legacy surveys (i.e., those with SAS summary scripts). 
 
 #### _ISSUES_
 1.  Work carefully through first few datasets, as the code is a bit coarse in places
 
 #### _TODO_
+1. New users should add their username and relative file path to 
+"user_lookup.csv" in this folder.
 
 ## Overhead
 
 #### Libraries
 ```{r overhead}
-library(knitr)
-library(reshape2)
-suppressMessages(library(dplyr))
-library(stringr)
+list.of.packages <- c(
+  "tidyverse",
+  "knitr"
+  )
+
+new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])]
+
+if(length(new.packages)) install.packages(new.packages)
+
+for (p in list.of.packages){
+  library(p, character.only = TRUE)
+}
 ```
 
 #### Knitr config
 ```{r config, include=FALSE}
-knitr::opts_chunk$set(cache=TRUE)
+knitr::opts_chunk$set(cache=TRUE, error = TRUE)
+suppressMessages(library(tidyverse))
+options(stringsAsFactors = FALSE)
 ```
 
 #### Remote file names
 ```{r file-names}
-F_SPATIAL_TO_BE_GEOCODED = "M:/Data/OnBoard/Data and Reports/_geocoding Standardized/spatial_places_to_be_geocoded.csv"
-F_BOARD_TO_BE_GEOCODED   = "M:/Data/OnBoard/Data and Reports/_geocoding Standardized/boarding_places_to_be_geocoded.csv"
-F_ALIGHT_TO_BE_GEOCODED  = "M:/Data/OnBoard/Data and Reports/_geocoding Standardized/alighting_places_to_be_geocoded.csv"
+user_list <- read.csv("user_lookup.csv")
+me <- Sys.getenv("USERNAME")
+dir_path <- user_list %>%
+  filter(user == me) %>%
+  .$path_switch
+
+f_spatial_to_be_geocoded_path = paste0(dir_path, "Data and Reports/_geocoding Standardized/spatial_places_to_be_geocoded.csv")
```

Comment:
Given that you have `Data and Reports` written over and over again, may it be better (i.e., easier to read) to assume this part of the path lives in `dir_path`?